yb-voyager/cmd/exportData.go

```diff
@@ -798,6 +807,200 @@ func getInitialTableList() (map[string]string, []sqlname.NameTuple) {
 	return partitionsToRootTableMap, finalTableList
 }
 
+func getInitialTableList() (map[string]string, []sqlname.NameTuple) {
+	// store table list after filtering unsupported or unnecessary tables
+	msr, err := metaDB.GetMigrationStatusRecord()
+	if err != nil {
+		utils.ErrExit("error fetching migration status record: %v", err)
+	}
+
+	if startClean || len(msr.TableListExportedFromSource) == 0 {
+		// fresh start case or the first run where we don't have a table list stored in msr
+		return fetchTablesNamesFromSourceAndFilterTableList()
+
+	}
+
+	//resumption case where we will use a table-list stored in
+	var fullTableList []sqlname.NameTuple
+	var finalTableList []sqlname.NameTuple
+	storedTableList := msr.TableListExportedFromSource
+	if source.DBType == POSTGRESQL && msr.SourceExportedTableListWithLeafPartitions != nil {
+		//list with partitioned root table and all leafs
+		storedTableList = msr.SourceExportedTableListWithLeafPartitions
+	}
+	for _, t := range storedTableList {
+		var tuple sqlname.NameTuple
+		var err error
+		switch source.DBType {
+		case POSTGRESQL:
+			defaultSchemaName, _ := getDefaultSourceSchemaName()
```

Comment:
I'm a little concerned about createing nameTuples without referring to NameRegistry for all tables just to handle the leaf partition case. 

Is there a way we can distinguish between the two, and use name registry for the normal tables and manually construct for the leaf partitions? can we use the partitionsToRootTableMap maybe?

Reply:
> I'm a little concerned about createing nameTuples without referring to NameRegistry for all tables just to handle the leaf partition case.

We are not creating nameTuples for all cases, its only the leaf tables. If you that `getNameTupleFromObjectName` function implementation, it first check whether this object has any parent if yes then only create rest use the name registry one.